src/checkmatelib/client.py

```diff
@@ -14,13 +16,19 @@ class CheckmateClient:
 
     MAX_URL_LENGTH = 2000
 
-    def __init__(self, host):
+    def __init__(self, host, api_key=None):
         """Initialise a client for contacting the Checkmate service.
 
         :param host: The host including scheme, for the Checkmate service
+        :param api_key: Secret to connect to the server.
+            Defaults to CHECKMATE_API_KEY env var if present.
         """
         self._host = host.rstrip("/")
 
+        self.api_key = api_key
+        if not self.api_key:
+            self.api_key = os.getenv("CHECKMATE_API_KEY")
```

Comment:
Huh, I don't think I knew about `getenv`, and that's handy. It's maybe a bit more convenient than `os.environ`. Easier to mock if you have to, as well